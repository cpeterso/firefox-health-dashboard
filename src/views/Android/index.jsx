/* eslint-disable react/no-array-index-key */
import React, { Component } from 'react';
import Grid from '@material-ui/core/Grid';
import DashboardPage from '../../components/DashboardPage';
import Section from '../../components/Section';
import BugzillaUrlContainer from '../../containers/BugzillaUrlContainer';
import BugzillaGraph from '../../containers/BugzillaGraph';
import NimbledroidSection from '../../containers/NimbledroidSection';
import PerfherderGraphContainer from '../../containers/PerfherderGraphContainer';
import RedashContainer from '../../containers/RedashContainer';
import SETTINGS from '../../settings';
import CONFIG from '../../utils/nimbledroid/config';
import TP6mAggregate from '../../components/TP6mAggregate';

class Android extends Component {
  render() {
    const nimbledroidSubTitle = `${
      CONFIG.packageIdLabels[CONFIG.baseProduct]
    } vs ${CONFIG.packageIdLabels[CONFIG.compareProduct]}`;

    return (
      <DashboardPage title="Android" subtitle="Release criteria">
        <div>
          <Grid container spacing={24}>
            <Grid item xs={6} key="bugzilla">
              <Section title="Bugzilla">
                <BugzillaUrlContainer
                  includeBugCount
                  queries={[
                    {
                      text: 'All GV Fenix MVP bugs',
                      parameters: {
                        whiteboard: '[geckoview:fenix:p',
                      },
                    },
                  ]}
                />
                <BugzillaGraph
                  queries={[
                    {
                      label: 'GV M2 bugs',
                      parameters: {
                        resolution: ['---', 'FIXED'],
                        whiteboard: '[geckoview:fenix:m2]',
                      },
                    },
                    {
                      label: 'GV M3 bugs',
                      parameters: {
                        resolution: ['---', 'FIXED'],
                        whiteboard: '[geckoview:fenix:m3]',
                      },
                    },
                    {
                      label: 'GV M4 bugs',
                      parameters: {
                        resolution: ['---', 'FIXED'],
                        whiteboard: '[geckoview:fenix:m4]',
                      },
                    },
                    {
                      label: 'GV M5 bugs',
                      parameters: {
                        resolution: ['---', 'FIXED'],
                        whiteboard: '[geckoview:fenix:m5]',
                      },
                    },
                    {
                      label: 'All GV Fenix MVP bugs',
                      parameters: {
                        resolution: ['---', 'FIXED'],
                        whiteboard: '[geckoview:fenix:m',
                      },
                    },
                  ]}
                  startDate="2019-02-01"
                  title="GeckoView Fenix bugs"
                />
              </Section>
            </Grid>

            <Grid item xs={6} key="nimbledroid">
              <Section title="Nimbledroid" subtitle={nimbledroidSubTitle}>
                <NimbledroidSection configuration={{ ...CONFIG }} />
              </Section>
            </Grid>
          </Grid>
        </div>
        <Section title="Raptor (TP6m)" more="/android/tp6m?test=loadtime">
          <TP6mAggregate />
        </Section>
        <Section title="Telemetry">
          <Grid container spacing={24}>
            <Grid item xs={6}>
              <RedashContainer
                title="Total content page load time (no 95th)"
                redashDataUrl="https://sql.telemetry.mozilla.org/api/queries/59395/results.json?api_key=2L0YcuUULtECr9bfew9OAEgtC50G4Ri8NCSPLR5F"
                redashQueryUrl="https://sql.telemetry.mozilla.org/queries/59395"
              />
            </Grid>
            <Grid item xs={6}>
              <RedashContainer
                title="Total content page load time"
                redashDataUrl="https://sql.telemetry.mozilla.org/api/queries/59397/results.json?api_key=u9eculhXgxqgsluxYGxfXaWQ6g7KCXioEvfwjK83"
                redashQueryUrl="https://sql.telemetry.mozilla.org/queries/59397"
              />
            </Grid>
          </Grid>
        </Section>
        <Section
          title="Perfherder"
          subtitle="Lower in the graph is better regardless if it is a score or execution time (read the Y label)">
          <Grid container spacing={24}>
            <Grid item xs={6}>
              <PerfherderGraphContainer
                title="Speedometer"
                series={[
                  {
                    label: 'Moto G5 (arm7)',
                    seriesConfig: {
                      framework: 10,
                      platform: 'android-hw-g5-7-0-arm7-api-16',
                      option: 'opt',
                      project: 'mozilla-central',
                      suite: 'raptor-speedometer-geckoview',
                    },
                  },
                  {
                    label: 'Pixel 2 (arm7)',
                    seriesConfig: {
                      framework: 10,
                      option: 'opt',
                      platform: 'android-hw-p2-8-0-arm7-api-16',
                      project: 'mozilla-central',
                      suite: 'raptor-speedometer-geckoview',
                    },
                  },
                  {
                    label: 'Pixel 2 (ARM64)',
                    seriesConfig: {
                      framework: 10,
                      option: 'opt',
                      platform: 'android-hw-p2-8-0-android-aarch64',
                      project: 'mozilla-central',
                      suite: 'raptor-speedometer-geckoview',
                    },
                  },
                  {
                    label: 'Moto G5 (pgo)',
                    seriesConfig: {
                      framework: 10,
                      platform: 'android-hw-g5-7-0-arm7-api-16-pgo',
                      project: 'mozilla-central',
                      suite: 'raptor-speedometer-geckoview',
                    },
                  },
                  {
                    label: 'Pixel 2 (pgo)',
                    seriesConfig: {
                      framework: 10,
                      platform: 'android-hw-p2-8-0-arm7-api-16-pgo',
                      project: 'mozilla-central',
                      suite: 'raptor-speedometer-geckoview',
                    },
                  },
                ]}
              />
            </Grid>
            <Grid item xs={6}>
              <PerfherderGraphContainer
                title="Unity WebGl"
                series={[
                  {
                    color: SETTINGS.colors[0],
                    label: 'Moto G5 (arm7)',
                    seriesConfig: {
                      framework: 10,
                      platform: 'android-hw-g5-7-0-arm7-api-16',
                      option: 'opt',
                      project: 'mozilla-central',
                      suite: 'raptor-unity-webgl-geckoview',
                    },
                  },
                  {
                    color: SETTINGS.colors[1],
                    label: 'Pixel 2 (arm7)',
                    seriesConfig: {
                      framework: 10,
                      platform: 'android-hw-p2-8-0-arm7-api-16',
                      option: 'opt',
                      project: 'mozilla-central',
                      suite: 'raptor-unity-webgl-geckoview',
                    },
                  },
                ]}
              />
            </Grid>
          </Grid>
        </Section>
      </DashboardPage>
    );
  }
}

export default Android;
