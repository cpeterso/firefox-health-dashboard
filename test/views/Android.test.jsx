import React from 'react';
import renderer from 'react-test-renderer';
import Android from '../../src/views/Android';

it('renders correctly', () => {
  const tree = renderer.create(<Android />).toJSON();

  expect(tree).toMatchSnapshot();
});
