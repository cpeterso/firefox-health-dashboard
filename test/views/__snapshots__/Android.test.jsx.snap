// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="DashboardPage-root-1"
>
  <div
    className="DashboardPage-title-2"
  >
    <h1
      style={
        Object {
          "display": "inline",
        }
      }
    >
      Android
      <small
        className="DashboardPage-subtitle-3"
      >
        Release criteria
      </small>
    </h1>
    <span
      style={
        Object {
          "float": "right",
        }
      }
    >
      <a
        href="https://github.com/mozilla-frontend-infra/firefox-health-dashboard/"
      >
        <img
          alt="Go to Github Repo"
          src="test-file-stub"
          style={
            Object {
              "padding": "0 0.5rem 0 0",
              "width": "2rem",
            }
          }
        />
      </a>
    </span>
  </div>
  <div>
    <div
      className="MuiGrid-container-4 MuiGrid-spacing-xs-24-28"
    >
      <div
        className="MuiGrid-item-5 MuiGrid-grid-xs-6-38"
      >
        <div>
          <div
            className="Section-header-102"
          >
            <h2
              className="Section-title-103"
            >
              <span>
                Bugzilla
              </span>
            </h2>
          </div>
          <div
            className="Section-content-101"
          >
            <div
              style={
                Object {
                  "lineHeight": "100%",
                  "textAlign": "center",
                  "width": "100%",
                }
              }
            >
              <div
                className="MuiCircularProgress-root-109 MuiCircularProgress-colorPrimary-112 MuiCircularProgress-indeterminate-111"
                role="progressbar"
                style={
                  Object {
                    "height": 40,
                    "width": 40,
                  }
                }
              >
                <svg
                  className="MuiCircularProgress-svg-114"
                  viewBox="22 22 44 44"
                >
                  <circle
                    className="MuiCircularProgress-circle-115 MuiCircularProgress-circleIndeterminate-117"
                    cx={44}
                    cy={44}
                    fill="none"
                    r={20.2}
                    strokeWidth={3.6}
                    style={Object {}}
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-item-5 MuiGrid-grid-xs-6-38"
      >
        <div>
          <div
            className="Section-header-102"
          >
            <h2
              className="Section-title-103"
            >
              <span>
                Nimbledroid
                <span
                  className="Section-subtitle-104"
                >
                  GeckoView Example vs Chrome Beta
                </span>
              </span>
            </h2>
          </div>
          <div
            className="Section-content-101"
          >
            <div />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="Section-header-102"
    >
      <h2
        className="Section-title-103"
      >
        <span>
          Page Load tests (TP6m)
          <span>
             (
            <a
              href="/android/tp6m?test=loadtime"
            >
              more
            </a>
            )
          </span>
        </span>
      </h2>
    </div>
    <div
      className="Section-content-101"
    >
      <div
        className="MuiGrid-container-4 MuiGrid-spacing-xs-24-28"
      >
        <div
          className="MuiGrid-item-5 MuiGrid-grid-xs-6-38"
        >
          <div>
            <h2
              className="ErrorBoundary-title-119"
            >
              <span>
                Tp6 mobile: Facebook
              </span>
            </h2>
            <div
              style={
                Object {
                  "lineHeight": "100%",
                  "textAlign": "center",
                  "width": "100%",
                }
              }
            >
              <div
                className="MuiCircularProgress-root-109 MuiCircularProgress-colorPrimary-112 MuiCircularProgress-indeterminate-111"
                role="progressbar"
                style={
                  Object {
                    "height": 40,
                    "width": 40,
                  }
                }
              >
                <svg
                  className="MuiCircularProgress-svg-114"
                  viewBox="22 22 44 44"
                >
                  <circle
                    className="MuiCircularProgress-circle-115 MuiCircularProgress-circleIndeterminate-117"
                    cx={44}
                    cy={44}
                    fill="none"
                    r={20.2}
                    strokeWidth={3.6}
                    style={Object {}}
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-item-5 MuiGrid-grid-xs-6-38"
        >
          <div>
            <h2
              className="ErrorBoundary-title-119"
            >
              <span>
                Tp6 mobile: Google
              </span>
            </h2>
            <div
              style={
                Object {
                  "lineHeight": "100%",
                  "textAlign": "center",
                  "width": "100%",
                }
              }
            >
              <div
                className="MuiCircularProgress-root-109 MuiCircularProgress-colorPrimary-112 MuiCircularProgress-indeterminate-111"
                role="progressbar"
                style={
                  Object {
                    "height": 40,
                    "width": 40,
                  }
                }
              >
                <svg
                  className="MuiCircularProgress-svg-114"
                  viewBox="22 22 44 44"
                >
                  <circle
                    className="MuiCircularProgress-circle-115 MuiCircularProgress-circleIndeterminate-117"
                    cx={44}
                    cy={44}
                    fill="none"
                    r={20.2}
                    strokeWidth={3.6}
                    style={Object {}}
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-item-5 MuiGrid-grid-xs-6-38"
        >
          <div>
            <h2
              className="ErrorBoundary-title-119"
            >
              <span>
                Tp6 mobile: Wikipedia
              </span>
            </h2>
            <div
              style={
                Object {
                  "lineHeight": "100%",
                  "textAlign": "center",
                  "width": "100%",
                }
              }
            >
              <div
                className="MuiCircularProgress-root-109 MuiCircularProgress-colorPrimary-112 MuiCircularProgress-indeterminate-111"
                role="progressbar"
                style={
                  Object {
                    "height": 40,
                    "width": 40,
                  }
                }
              >
                <svg
                  className="MuiCircularProgress-svg-114"
                  viewBox="22 22 44 44"
                >
                  <circle
                    className="MuiCircularProgress-circle-115 MuiCircularProgress-circleIndeterminate-117"
                    cx={44}
                    cy={44}
                    fill="none"
                    r={20.2}
                    strokeWidth={3.6}
                    style={Object {}}
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-item-5 MuiGrid-grid-xs-6-38"
        >
          <div>
            <h2
              className="ErrorBoundary-title-119"
            >
              <span>
                Tp6 mobile: YouTube
              </span>
            </h2>
            <div
              style={
                Object {
                  "lineHeight": "100%",
                  "textAlign": "center",
                  "width": "100%",
                }
              }
            >
              <div
                className="MuiCircularProgress-root-109 MuiCircularProgress-colorPrimary-112 MuiCircularProgress-indeterminate-111"
                role="progressbar"
                style={
                  Object {
                    "height": 40,
                    "width": 40,
                  }
                }
              >
                <svg
                  className="MuiCircularProgress-svg-114"
                  viewBox="22 22 44 44"
                >
                  <circle
                    className="MuiCircularProgress-circle-115 MuiCircularProgress-circleIndeterminate-117"
                    cx={44}
                    cy={44}
                    fill="none"
                    r={20.2}
                    strokeWidth={3.6}
                    style={Object {}}
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="Section-header-102"
    >
      <h2
        className="Section-title-103"
      >
        <span>
          Telemetry
        </span>
      </h2>
    </div>
    <div
      className="Section-content-101"
    >
      <div
        style={
          Object {
            "display": "flex",
            "flexWrap": "wrap",
          }
        }
      >
        <div>
          <div
            style={
              Object {
                "lineHeight": "100%",
                "textAlign": "center",
                "width": "100%",
              }
            }
          >
            <div
              className="MuiCircularProgress-root-109 MuiCircularProgress-colorPrimary-112 MuiCircularProgress-indeterminate-111"
              role="progressbar"
              style={
                Object {
                  "height": 40,
                  "width": 40,
                }
              }
            >
              <svg
                className="MuiCircularProgress-svg-114"
                viewBox="22 22 44 44"
              >
                <circle
                  className="MuiCircularProgress-circle-115 MuiCircularProgress-circleIndeterminate-117"
                  cx={44}
                  cy={44}
                  fill="none"
                  r={20.2}
                  strokeWidth={3.6}
                  style={Object {}}
                />
              </svg>
            </div>
          </div>
          <div
            className="ErrorBoundary-linkContainer-121"
          >
            <a
              href="https://sql.telemetry.mozilla.org/queries/59395"
              rel="noopener noreferrer"
              target="_blank"
            >
              <span
                className="ErrorBoundary-middleVerticalAlignment-122"
              >
                Redash query
              </span>
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-123 ErrorBoundary-middleVerticalAlignment-122"
                focusable="false"
                role="presentation"
                style={
                  Object {
                    "height": "1rem",
                  }
                }
                viewBox="0 0 24 24"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
                />
              </svg>
            </a>
          </div>
        </div>
        <div>
          <div
            style={
              Object {
                "lineHeight": "100%",
                "textAlign": "center",
                "width": "100%",
              }
            }
          >
            <div
              className="MuiCircularProgress-root-109 MuiCircularProgress-colorPrimary-112 MuiCircularProgress-indeterminate-111"
              role="progressbar"
              style={
                Object {
                  "height": 40,
                  "width": 40,
                }
              }
            >
              <svg
                className="MuiCircularProgress-svg-114"
                viewBox="22 22 44 44"
              >
                <circle
                  className="MuiCircularProgress-circle-115 MuiCircularProgress-circleIndeterminate-117"
                  cx={44}
                  cy={44}
                  fill="none"
                  r={20.2}
                  strokeWidth={3.6}
                  style={Object {}}
                />
              </svg>
            </div>
          </div>
          <div
            className="ErrorBoundary-linkContainer-121"
          >
            <a
              href="https://sql.telemetry.mozilla.org/queries/59397"
              rel="noopener noreferrer"
              target="_blank"
            >
              <span
                className="ErrorBoundary-middleVerticalAlignment-122"
              >
                Redash query
              </span>
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-123 ErrorBoundary-middleVerticalAlignment-122"
                focusable="false"
                role="presentation"
                style={
                  Object {
                    "height": "1rem",
                  }
                }
                viewBox="0 0 24 24"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="Section-header-102"
    >
      <h2
        className="Section-title-103"
      >
        <span>
          Perfherder
          <span
            className="Section-subtitle-104"
          >
            Lower in the graph is better regardless if it is a score or execution time (read the Y label)
          </span>
        </span>
      </h2>
    </div>
    <div
      className="Section-content-101"
    >
      <div
        style={
          Object {
            "display": "flex",
            "flexWrap": "wrap",
          }
        }
      >
        <div>
          <h2
            className="ErrorBoundary-title-119"
          >
            <span>
              Speedometer
            </span>
          </h2>
          <div
            style={
              Object {
                "lineHeight": "100%",
                "textAlign": "center",
                "width": "100%",
              }
            }
          >
            <div
              className="MuiCircularProgress-root-109 MuiCircularProgress-colorPrimary-112 MuiCircularProgress-indeterminate-111"
              role="progressbar"
              style={
                Object {
                  "height": 40,
                  "width": 40,
                }
              }
            >
              <svg
                className="MuiCircularProgress-svg-114"
                viewBox="22 22 44 44"
              >
                <circle
                  className="MuiCircularProgress-circle-115 MuiCircularProgress-circleIndeterminate-117"
                  cx={44}
                  cy={44}
                  fill="none"
                  r={20.2}
                  strokeWidth={3.6}
                  style={Object {}}
                />
              </svg>
            </div>
          </div>
        </div>
        <div>
          <h2
            className="ErrorBoundary-title-119"
          >
            <span>
              Unity WebGl
            </span>
          </h2>
          <div
            style={
              Object {
                "lineHeight": "100%",
                "textAlign": "center",
                "width": "100%",
              }
            }
          >
            <div
              className="MuiCircularProgress-root-109 MuiCircularProgress-colorPrimary-112 MuiCircularProgress-indeterminate-111"
              role="progressbar"
              style={
                Object {
                  "height": 40,
                  "width": 40,
                }
              }
            >
              <svg
                className="MuiCircularProgress-svg-114"
                viewBox="22 22 44 44"
              >
                <circle
                  className="MuiCircularProgress-circle-115 MuiCircularProgress-circleIndeterminate-117"
                  cx={44}
                  cy={44}
                  fill="none"
                  r={20.2}
                  strokeWidth={3.6}
                  style={Object {}}
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
